enhancements/cnf-tests-split.md

```diff
@@ -0,0 +1,204 @@
+ 1. Add the k8s reporter to each repository that doesn't has it already.
+ 2. Use ginkgo's built-in junit generator, so that we don't have to add the `-junit` flag to each repo.
+ 3. Create a local junit-merger program to aggregate all of the junit reports under the same test suite.
+ 4. Change the reports output directory. Having `TESTS_REPORTS_PATH` as the root path,
+   we will have to include "/raw" subpath, where we create all the junit reports, 
+   and a "/junit" subpath where we hold the merged junit reports.
+
+### Rewrite the cnf-tests container
+
+In the past Red Hat delivered the various cnf-tests as a product,
+however today we support only the latency tests within it.
+see [Performing latency tests for platform verification](https://docs.openshift.com/container-platform/4.13/scalability_and_performance/cnf-performing-platform-verification-latency-tests.html)
+
+In this refactoring we include changes to the cnf-tests container because
+currently the interface for the latency tests is not user-friendly.
+e.g. we currently tell the users to run it by:
+```bash
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e LATENCY_TEST_RUN=true -e DISCOVERY_MODE=true -e FEATURES=performance registry.redhat.io/openshift4/cnf-tests-rhel8:v4.13 \
+/usr/bin/test-run.sh -ginkgo.focus="\[performance\]\ Latency\ Test"
+```
+
+In the command above the inputs: `LATENCY_TEST_RUN`, `DISCOVERY_MODE`, `FEATURES`, and the `-ginkgo.focus`.
+can be hidden from the user if we write the container image to include only the latency tests, making a cleaner entrypoint.
+Moreover, the current container can be disruptive if you don't pass the
+`DISCOVERY_MODE` var or misspell it.  
+Because when it's not set to true,
+the binary will apply a different configuration to the cluster and may
+cause a reboot - something severe and not acceptable on the client's production cluster.
+
+The rewrite will alter the cnf-tests's Dockerfile, the `cnf-tests/entrypoint/test-run.sh` script
+and the `build-test-bin.sh` hack script. Making the container image include only the latency
+tests, and hardcoding the required parameters into the scripts.
+The documentation for this product will be updated to reflect the changes.
+### API Extensions
+N/A
+### Risks and Mitigations
+
+The most major risk here is revolving around changing the cnf-tests container.  
+That is because it's delivered as a product supported by Red Hat, and
+we should make sure to keep backward compatibility and provide the relevant documentation, explaining the new interface to the latency tests.  
+We should also provide a mitigation guide for the various internal users 
+for the cnf-tests running it on different pipelines.
+
+### Drawbacks
+
+In order to implement this we will have to touch other code bases and
+add the k8s reporter, which can be rejected/unwelcomed by the upstream community.  
+Also, the k8s reporter flag we introduce in the external repositories 
```

Comment:
This is only if  we invoke with the reporter flag, right?

Reply:
right. which we currently do.